src/fft/fftn.rs

```diff
@@ -0,0 +1,534 @@
+///   with zeros to match `n`. The default value is `a.shape[axis]`.
+/// - `axis`: Axis along which to perform the FFT. The default is -1.
+#[default_device(device = "cpu")]
+pub unsafe fn fft_device_unchecked(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Array {
+    let (n, axis) = resolve_size_and_axis_unchecked(a, n, axis);
+    unsafe {
+        let c_array = mlx_sys::mlx_fft_fft(a.c_array, n, axis, stream.stream.c_stream);
+        Array::from_ptr(c_array)
+    }
+}
+
+/// One dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///   with zeros to match `n`. The default value is `a.shape[axis]`.
+/// - `axis`: Axis along which to perform the FFT. The default is -1.
+#[default_device(device = "cpu")]
+pub fn try_fft_device(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Result<Array, FftError> {
+    let (n, axis) = try_resolve_size_and_axis(a, n, axis)?;
+    unsafe { Ok(fft_device_unchecked(a, Some(n), Some(axis), stream)) }
+}
+
+/// One dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///   with zeros to match `n`. The default value is `a.shape[axis]`.
+/// - `axis`: Axis along which to perform the FFT. The default is -1.
+///
+/// # Panic
+///
+/// Panics if the input array is a scalar or if the axis is invalid.
+///
+/// See [`try_fft_device`] for more details.
+#[default_device(device = "cpu")]
+pub fn fft_device(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Array {
+    try_fft_device(a, n, axis, stream).unwrap()
+}
+
+/// Two dimensional discrete Fourier Transform.
+///
+/// # Param
+///
+/// - `a`: The input array.
+/// - `s`: Size of the transformed axes. The corresponding axes in the input are truncated or padded
+///  with zeros to match `n`. The default value is the sizes of `a` along `axes`.
+/// - `axes`: Axes along which to perform the FFT. The default is `[-2, -1]`.
+#[default_device(device = "cpu")]
+pub unsafe fn fft2_device_unchecked<'a>(
+    a: &'a Array,
+    s: impl Into<Option<&'a [i32]>>,
+    axes: impl Into<Option<&'a [i32]>>,
+    stream: StreamOrDevice,
+) -> Array {
+    let axes = axes.into().unwrap_or(&[-2, -1]);
+    fftn_device_unchecked(a, s, axes, stream)
+}
+
+/// Two dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `s`: Size of the transformed axes. The corresponding axes in the input are truncated or padded
+/// with zeros to match `n`. The default value is the sizes of `a` along `axes`.
```

Comment:
```suggestion
/// with zeros to match `s`. The default value is the sizes of `a` along `axes`.
```

Reply:
This follows the python api. 1D fft uses n, n-D uses s

Reply:
In this case though, you used `s` as a name but then used `n` here. so there's a mismatch

Reply:
Yeah, didn't realize that